Destroy big iq license causing error

### Environment

 * TMOS/Bigip Version: 15.1.4
 * Terraform Version: v1.1.6
 * Terraform bigip provider Version: 1.12.2

### Summary
The big iq license module works really well when creating a license. However when destroying the object I get this error:

│ Error: strconv.Atoi: parsing "": invalid syntax
│ 
│ 

The destroy doesnt remove the license, for this I wrote a little script, but what is worse is that the terraform destroy hangs on this object not being deleted. so the rest of my infrastructure doesnt get deleted. (terragrunt fails to continue if not all objects are deleted in the previous step)

### Steps To Reproduce

Steps to reproduce the behavior:

1. Provide terraform resource config which you are facing trouble along with the output of it.

resource "bigip_common_license_manage_bigiq" "bigip_license" {
  bigiq_address    = var.bigiq_ip
  bigiq_user       = var.bigiq_username
  bigiq_password   = var.bigiq_password
  license_poolname = local.license_name
  skukeyword1      = local.license_type
  assignment_type  = "UNMANAGED"
  unit_of_measure  = "yearly"
}

2. To get to know more about the issue, provide terraform debug logs
```
2022-02-22T17:22:07.126-0300 [DEBUG] provider.terraform-provider-bigip_v1.12.2: 2022/02/22 17:22:07 Revoke License assignment for :https://1.1.1.1
2022-02-22T17:22:07.126-0300 [DEBUG] provider.terraform-provider-bigip_v1.12.2: 2022/02/22 17:22:07 [INFO] Initializing BigIP connection
2022-02-22T17:22:07.330-0300 [ERROR] vertex "module.f5.bigip_common_license_manage_bigiq.bigip_license (destroy)" error: strconv.Atoi: parsing "": invalid syntax
2022-02-22T17:22:09.348-0300 [DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = transport is closing"
2022-02-22T17:22:09.350-0300 [DEBUG] provider: plugin exited
╷
│ Error: strconv.Atoi: parsing "": invalid syntax
│ 
│ 
╵
2022-02-22T17:22:09.364-0300 [DEBUG] provider: plugin process exited: path=.terraform/providers/registry.terraform.io/f5networks/bigip/1.12.2/linux_amd64/terraform-provider-bigip_v1.12.2 pid=8846
2022-02-22T17:22:09.364-0300 [DEBUG] provider: plugin exited
```

(IP has been changed)
### Expected Behavior
When executing a terraform destroy all the components get deleted and in this case the license gets revoked.

### Actual Behavior
The license is added correctly, however the destroy does not work and ends with an Error: strconv.Atoi: parsing "": invalid syntax.

today during the destroy phase, I received more errors:
``` 
╷
│ Error: strconv.Atoi: parsing "": invalid syntax
│ 
│ 
╵
╷
│ Error: Plugin did not respond
│ 
│ The plugin encountered an error, and failed to respond to the
│ plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may
│ contain more details.
╵

Stack trace from the terraform-provider-bigip_v1.12.2 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x73a88a]

goroutine 70 [running]:
github.com/f5devcentral/go-bigip.(*BigIP).DeleteAs3Bigip(0xc0005f00e0, 0xc00003e852, 0xa, 0x0, 0x0, 0x30, 0xc0007eea80)
        github.com/f5devcentral/go-bigip@v0.0.0-20220119082918-5f215f878998/as3bigip.go:148 +0x28a
github.com/F5Networks/terraform-provider-bigip/bigip.resourceBigipAs3Delete(0xc000140b60, 0x10262e0, 0xc0005f00e0, 0x0, 0x0)
        github.com/F5Networks/terraform-provider-bigip/bigip/resource_bigip_as3.go:367 +0x12e
github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Resource).Apply(0xc000592080, 0xc00047e960, 0xc0006764c0, 0x10262e0, 0xc0005f00e0, 0x40e001, 0x20, 0xfb6aa0)
        github.com/hashicorp/terraform-plugin-sdk@v1.1.0/helper/schema/resource.go:278 +0x4a6
github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Provider).Apply(0xc000592880, 0xc0001138e8, 0xc00047e960, 0xc0006764c0, 0x175f400, 0x0, 0x0)
        github.com/hashicorp/terraform-plugin-sdk@v1.1.0/helper/schema/provider.go:294 +0x99
github.com/hashicorp/terraform-plugin-sdk/internal/helper/plugin.(*GRPCProviderServer).ApplyResourceChange(0xc0001a21b0, 0x11ba050, 0xc000457fb0, 0xc0006da9c0, 0xc0001a21b0, 0xc000457fb0, 0xc0003a9a50)
        github.com/hashicorp/terraform-plugin-sdk@v1.1.0/internal/helper/plugin/grpc_provider.go:885 +0x8a5
github.com/hashicorp/terraform-plugin-sdk/internal/tfplugin5._Provider_ApplyResourceChange_Handler(0xfe5160, 0xc0001a21b0, 0x11ba050, 0xc000457fb0, 0xc0006da960, 0x0, 0x11ba050, 0xc000457fb0, 0xc00027e000, 0x2c6)
        github.com/hashicorp/terraform-plugin-sdk@v1.1.0/internal/tfplugin5/tfplugin5.pb.go:3189 +0x214
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0000d2160, 0x11c2838, 0xc000102c00, 0xc000136100, 0xc000189b60, 0x171fca0, 0x0, 0x0, 0x0)
        google.golang.org/grpc@v1.23.0/server.go:995 +0x482
google.golang.org/grpc.(*Server).handleStream(0xc0000d2160, 0x11c2838, 0xc000102c00, 0xc000136100, 0x0)
        google.golang.org/grpc@v1.23.0/server.go:1275 +0xd2c
google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc0001201c0, 0xc0000d2160, 0x11c2838, 0xc000102c00, 0xc000136100)
        google.golang.org/grpc@v1.23.0/server.go:710 +0xab
created by google.golang.org/grpc.(*Server).serveStreams.func1
        google.golang.org/grpc@v1.23.0/server.go:708 +0xa5

Error: The terraform-provider-bigip_v1.12.2 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```
It seems like a bug in the provider code, would anyone be able to help me with this issue?

--- a/bigip/resource_bigiq_regkey_license_manage.go
+++ b/bigip/resource_bigiq_regkey_license_manage.go
@@ -131,7 +131,10 @@ func resourceBigiqLicenseManageCreate(d *schema.ResourceData, meta interface{}) 
 	var deviceIP []string
 	var respID string
 	deviceIP = getDeviceUri(bigipRef.Host)
-	devicePort, _ := strconv.Atoi(deviceIP[3])
+	var devicePort int
+	if port, err := strconv.Atoi(deviceIP[3]); err == nil {
+		devicePort = port
+	}
 	licensePoolName := d.Get("license_poolname").(string)
 	log.Printf("[INFO] BIGIP License Assignment Started on Pool:%v", licensePoolName)
 	poolInfo, err := bigiqRef.GetPoolType(licensePoolName)
@@ -314,7 +317,10 @@ func resourceBigiqLicenseManageUpdate(d *schema.ResourceData, meta interface{}) 
 	var deviceIP []string
 	var respID string
 	deviceIP = getDeviceUri(bigipRef.Host)
-	devicePort, _ := strconv.Atoi(deviceIP[3])
+	var devicePort int
+	if port, err := strconv.Atoi(deviceIP[3]); err == nil {
+		devicePort = port
+	}
 	licensePoolName := d.Get("license_poolname").(string)
 	poolInfo, err := bigiqRef.GetPoolType(licensePoolName)
 	if err != nil {
@@ -426,9 +432,9 @@ func resourceBigiqLicenseManageDelete(d *schema.ResourceData, meta interface{}) 
 	}
 
 	deviceIP := getDeviceUri(bigipRef.Host)
-	devicePort, err := strconv.Atoi(deviceIP[3])
-	if err != nil {
-		return err
+	var devicePort int
+	if port, err := strconv.Atoi(deviceIP[3]); err == nil {
+		devicePort = port
 	}
 	assignmentType := d.Get("assignment_type").(string)
 	if regKey == "" {
@@ -474,13 +480,13 @@ func resourceBigiqLicenseManageDelete(d *schema.ResourceData, meta interface{}) 
 			time.Sleep(5 * time.Second)
 		}
 		log.Println("[DEBUG] wait for bigip status with license revoking")
-		bigipLicence, err := bigipRef.GetBigipLiceseStatus()
+		bigipLicence, err := waitLicenseRevoke(bigipRef)
 		if err != nil {
 			return fmt.Errorf("getting license revoking status from bigip failed with :%v", err)
 		}
 		_, ok := bigipLicence["entries"].(map[string]interface{})
 		if ok {
-			return fmt.Errorf("getting license revoking status from bigip failed with :%v", err)
+			return fmt.Errorf("getting license revoking status from bigip failed")
 		}
 		log.Printf("[INFO] License Revoking for Device %+v Success", bigipRef.Host)
 	} else {
@@ -506,6 +512,19 @@ func resourceBigiqLicenseManageDelete(d *schema.ResourceData, meta interface{}) 
 	}
 	d.SetId("")
 	return nil
+}
+
+func waitLicenseRevoke(bigipRef *bigip.BigIP) (map[string]interface{}, error) {
+	bigipLicense, err := bigipRef.GetBigipLiceseStatus()
+	if err != nil {
+		return nil, err
+	}
+	retries := 0
+	for _, ok := bigipLicense["entries"]; ok && retries < 3; retries += 1 {
+		time.Sleep(time.Second * 5)
+		bigipLicense, err = bigipRef.GetBigipLiceseStatus()
+	}
+	return bigipLicense, err
 }
 
 func connectBigIq(d *schema.ResourceData) (*bigip.BigIP, error) {

